### PR TITLE
Saved favorites modification

### PR DESCRIPTION
**Pull request by codingAlchemist**

### CODE REVIEW

File: OceanCandy/StateListViewController.m [hunk 1]
```diff
@@ -64,7 +57,8 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
 }
 -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
     StationListViewController *stationList = [self.storyboard instantiateViewControllerWithIdentifier:@"stationList"];
-    [stationList setStations:[self.stations objectForKey:[[self orderedStates] objectAtIndex:indexPath.row]]];
+    StateListCell *cell = (StateListCell *)[tableView cellForRowAtIndexPath:indexPath];
+    [stationList setStations:[self.stations objectForKey:cell.nameLabel.text]];
```
Comment (greis) [inline OceanCandy/StateListViewController.m]: Why this change? You should try to avoid asking a cell for some data. And even if this is the only way to do this, you should not access any ui elements of the cell. It would be better if you create a method on the cell that returns the station name: `[cell stationName]`

File: OceanCandy/Tide.h [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 #import <Foundation/Foundation.h>
 #import <CoreData/CoreData.h>
 
+@class Station;
 
 @interface Tide : NSManagedObject
 
-@property (nonatomic, retain) NSString * time;
 @property (nonatomic, retain) NSString * feet;
+@property (nonatomic, retain) NSString * time;
```
Comment (greis) [inline OceanCandy/Tide.h]: Don't forget to always put the pointer together with the variable

File: OceanCandy/TideInfoViewController.m [hunk 3]
```diff
@@ -1,56 +1,72 @@
...
                 }];
             }];
+            
         }];
-    }];
+    }else{
+        self.addToFavoritesButton.hidden = YES;
+        NSSet *tides = self.station.tides;
+        NSArray *tideArray = [tides allObjects];
+        self.firstHighFeetLabel.text = [[tideArray objectAtIndex:0]valueForKey:@"feet"];
```
Comment (greis) [inline OceanCandy/TideInfoViewController.m]: It would be nice to use the new syntax for arrays and dictionaries: `tideArray[0][@"feet"]`

File: OceanCandy/StateListViewController.m (hunk 1 above)
Comment (codingAlchemist) [inline OceanCandy/StateListViewController.m]: I will try and implement that now and then ill resubmit

File: OceanCandy/Tide.h (hunk 2 above)
Comment (codingAlchemist) [inline OceanCandy/Tide.h]: auto generated by xcode when making a new NSObjectContext -- will fix

File: OceanCandy/TideInfoViewController.m (hunk 3 above)
Comment (codingAlchemist) [inline OceanCandy/TideInfoViewController.m]: Thanks sometimes i forget that you can do that, old habits and such. Ill try and remember to do this from now on :)

File: OceanCandy/StateListViewController.m (hunk 1 above)
Comment (codingAlchemist) [inline OceanCandy/StateListViewController.m]: instead of asking the ui or creating a new method i just used the array i was already using to get station selected by [self orderedStates][indexPath.row] <-- see what i did there :)
